src/CommandFacility.cpp

```diff
@@ -41,30 +41,29 @@ CommandFacility::set_commanded(CommandedObject& commanded)
 }
 
 void 
-CommandFacility::execute_command(const cmdobj_t& command)
+CommandFacility::execute_command(cmdmeta_t meta)
 {
-  auto execfut = std::async(std::launch::deferred, m_command_callback, std::move(command));
+  auto execfut = std::async(std::launch::deferred, m_command_callback, std::move(meta));
   m_completion_queue.push(std::move(execfut));
 }
 
 void
-CommandFacility::handle_command(const cmdobj_t& command) 
+CommandFacility::handle_command(cmdmeta_t meta)
 {
-  std::string ret = "";
   try {
-    m_commanded_object->execute(command);
-    ret = "OK";
+    m_commanded_object->execute(nlohmann::json::parse(meta["command"]));
```

Comment:
Yeah, squashing together the meta concept with the `cmdobj` type is probably not good due to the multiple copies and parses required.

Reply:
This should be fixed (see other).